pkg/cloudprovider/providers/gce/gce_addresses_fakes.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"net/http"
+	"strconv"
+
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+type FakeCloudAddressService struct {
+	count    int
+	reserved map[string]bool
+	m        map[string]map[string]*compute.Address
+}
+
+func NewFakeCloudAddressService() *FakeCloudAddressService {
+	return &FakeCloudAddressService{
+		reserved: make(map[string]bool),
+		m:        make(map[string]map[string]*compute.Address),
+	}
+}
+
+func (cas *FakeCloudAddressService) ReserveRegionAddress(addr *compute.Address, region string) error {
+	if addr.Address == "" {
+		addr.Address = strconv.Itoa(cas.count)
```

Comment:
would prefer fmt.Sprintf("1.2.3.%v", cas.count) to render a more familiar IP address